src/null-safety/index.md

```diff
@@ -3,13 +3,31 @@ title: Sound null safety
 description: Information about Dart's null safety feature
 ---
 
-The Dart language now supports sound null safety!
+The Dart language comes with sound null safety!
 
-When you opt into null safety,
+When using sound null safety,
 types in your code are non-nullable by default, meaning that
 variables can’t contain `null` _unless you say they can._
 With null safety, your **runtime** null-dereference errors
-turn into **edit-time** analysis errors.
+turn into **edit-time** analysis errors, enabling you to fix
+errors before deploying to customers.
+
+{{site.alert.warn}}
+In Dart 2.x SDKs, sound null safety is enabled or disabled
+via configuration of the project SDK constraint;
+see [Enabling/disabling null safety](#enable-null-safety) for details.
+
+In Dart 3 -- tentatively planned for release by mid 2023 --
+sound null safety will always be enabled,
+and running _without_ sound null will no longer be possible.
+All existing code must be [migrated](#migrate) to sound null safety
+to be compatible with Dart 3.
+For details, see the [Dart 3 sound null safety tracking issue][].
+{{site.alert.end}}
+
+[Dart 3 sound null safety tracking issue]: https://github.com/dart-lang/sdk/issues/49530
+
+## Introduction via examples
```

Comment:
We tend to not use [via](https://developers.google.com/style/word-list#via). Perhaps "through" would be better?

```suggestion
## Introduction through examples
```